### PR TITLE
feat: add shell syntax "commands"

### PR DESCRIPTION
**Pull request by SeparateRecords**
Fixes https://github.com/withfig/fig/issues/1317
Also adds support for basic things like `if` (and `then`/`elif`/`else`), `for`, etc.

### CODE REVIEW

File: src/where.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+const completionSpec: Fig.Spec = {
+  name: "whence",
+  description: "For each name, indicate how it should be interpreted",
```
Comment (mschrage) [inline src/where.ts:3]: Seems like `name` is incorrect?

File: src/which.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+const completionSpec: Fig.Spec = {
+  name: "whence",
+  description: "For each name, indicate how it should be interpreted",
```
Comment (mschrage) [inline src/which.ts:3]: Same thing here! File is `which.ts` but name is `whence`

File: src/where.ts (hunk 1 above)
Comment (SeparateRecords) [reply src/where.ts]: Fixed!

**Review by fedeci — APPROVED** [review]
🚀

**Pull request merged**